### PR TITLE
Identityintegration

### PR DESCRIPTION
**Pull request by dtracers**
Integrates the identity server into course sketch.

Depends on the merge request from #467

### CODE REVIEW

File: CourseSketchProjects/CourseSketchLoginServer/src/main/java/connection/LoginServerWebSocketHandler.java [hunk 1]
```diff
@@ -136,15 +140,14 @@ private void registerUserMessage(final SocketSession conn, final Request req, fi
...
-
             // login user after registering user.
             loginUser(conn, req, login);
         } catch (GeneralSecurityException | AuthenticationException e) {
             final Message.ProtoException protoEx = ExceptionUtilities.createProtoException(e);
             conn.send(ExceptionUtilities.createExceptionRequest(req, protoEx));
             LOG.error(LoggingConstants.EXCEPTION_MESSAGE, e);
             send(conn, createLoginResponse(req, login, false, e.getMessage(), null));
-        } catch (RegistrationException e) {
+        } catch (RegistrationException | DatabaseAccessException e) {
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchLoginServer/src/main/java/connection/LoginServerWebSocketHandler.java]: combine this catch with the previous catch cause you do the same thing in both

File: CourseSketchProjects/CourseSketchLoginServer/src/main/java/coursesketch/database/DatabaseClient.java [hunk 2]
```diff
@@ -234,10 +247,30 @@ private BasicDBObject getUserInfo(final DBObject cursor, final boolean loginAsDe
         } else {
             throw new LoginException(LoginServerWebSocketHandler.PERMISSION_ERROR_MESSAGE);
         }
+        final String userId = getUserId(cursor.get(DatabaseStringConstants.USER_NAME).toString(),
+                cursor.get(DatabaseStringConstants.IDENTITY_AUTH).toString());
+        result.append(DatabaseStringConstants.USER_ID, userId);
+        // gets user id
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchLoginServer/src/main/java/coursesketch/database/DatabaseClient.java]: move this comment 2 lines up

File: CourseSketchProjects/CourseSketchProxyServer/src/main/java/connection/LoginConnectionState.java [hunk 3]
```diff
@@ -48,16 +61,17 @@ public final boolean isLoggedIn() {
 
     /**
      * Logs in the connection.
-     *
-     * @param instructorFlag
+     *  @param instructorFlag
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchProxyServer/src/main/java/connection/LoginConnectionState.java]: remove ' ' before the '@'

File: CourseSketchProjects/CourseSketchProxyServer/src/main/java/connection/LoginConnectionState.java [hunk 4]
```diff
@@ -82,7 +96,15 @@ public final boolean isInstructor() {
...
 
     /**
-     * @return the session id of the user who logged in (its user id)
+     * @return the authentication id of the user who logged in (its auth id)
+     */
+    protected final String getServerAuthId() {
+        return serverAuthId;
+    }
+
+
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchProxyServer/src/main/java/connection/LoginConnectionState.java]: remove this line

File: CourseSketchProjects/CourseSketchProxyServer/src/main/java/coursesketch/serverfront/ProxyConnectionState.java [hunk 5]
```diff
@@ -18,9 +18,16 @@ public ProxyConnectionState(final String inputKey) {
...
      */
-    /* package-private */String getUserId() {
+    /* package-private */ String getAuthId() {
+        return getServerAuthId();
+    }
+
+    /**
+     * @return the identification id of the user who logged in (its user id)
+     */
+    /* package-private */ String getUserId() {
```
Comment (areftw) [inline CourseSketchProjects/CourseSketchProxyServer/src/main/java/coursesketch/serverfront/ProxyConnectionState.java]: is this comment supposed to be here?

File: CourseSketchProjects/ProtoFiles/src/main/proto/message.proto [hunk 6]
```diff
@@ -13,8 +13,9 @@ message Request {
     optional bytes otherData = 3;
     optional string responseText = 4;
     optional string sessionInfo = 5; // This may be separated out more.
-    optional string serversideId = 6;
+    optional string serversideId = 6; // This is the authenticationId
     optional int64 messageTime = 7; // this is the time that the message is received by the server
```
Comment (areftw) [inline CourseSketchProjects/ProtoFiles/src/main/proto/message.proto]: Capitalize 't' in this

File: CourseSketchProjects/rpcserver/src/main/java/coursesketch/identity/IdentityWebSocketClient.java [hunk 7]
```diff
@@ -0,0 +1,283 @@
...
+            final Identity.IdentityCreationRequest creationRequest = creationRequestBuilder.build();
+            response = identityService.createNewItem(getNewRpcController(), creationRequest);
+            if (response.hasException()) {
+                final AuthenticationException authExcep =
+                        new AuthenticationException("Exception with authentication server", AuthenticationException.OTHER);
+                authExcep.setProtoException(response.getException());
+                throw authExcep;
+            }
+        } catch (ServiceException e) {
+            throw new AuthenticationException(e);
```
Comment (areftw) [inline CourseSketchProjects/rpcserver/src/main/java/coursesketch/identity/IdentityWebSocketClient.java]: you don't print the stack trace here?

File: CourseSketchProjects/CourseSketchProxyServer/src/main/java/coursesketch/serverfront/ProxyConnectionState.java (hunk 5 above)
Comment (dtracers) [inline CourseSketchProjects/CourseSketchProxyServer/src/main/java/coursesketch/serverfront/ProxyConnectionState.java]: yes.  it lets you know that only package access is allowed.  (To make sure it is not a mistake)

**Pull request merged**